consensus/polybft/fsm.go

```diff
@@ -158,21 +171,46 @@ func (f *fsm) BuildProposal() (*pbft.Proposal, error) {
 		headerTime = time.Now()
 	}
 
+	currentValidatorsHash, err := f.validators.Accounts().Hash()
+	if err != nil {
+		return nil, err
+	}
+
+	nextValidatorsHash, err := nextValidators.Hash()
```

Comment:
Is nextValidators empty for non-end-of-epoch blocks?

Reply:
Yes it is:
https://github.com/0xPolygon/polygon-edge/pull/840/files#diff-e7ab7b5e6b57d0319949e67e49d31398cccb3e4625ff1d69a1403c1db6ba68c4R121.

Btw should we return `ZeroHash` or still calculate keccak256 of abi encoded empty tuple of validators?

Reply:
I thought to make currentValidatorsHash== nextValidatorsHash for the blocks without validator set changes.  Do you see any benefits/drawbacks?

Reply:
Don't have problem with that, but hopefully smart contract is going to handle it properly...?

Reply:
Fix [87de519](https://github.com/0xPolygon/polygon-edge/pull/840/commits/87de5193b197f89d24eb7f4644ca2a2a769ec43b)